src/libmongoc/doc/learn/get/from-source.rst

```diff
@@ -0,0 +1,386 @@
+
+With the source directory for |mongo-c-driver| at `$SOURCE` and build directory
+`$BUILD`, the following command can be executed from a command-line to configure
+the project with both |libbson| and |libmongoc|::
+
+  $ cmake -S $SOURCE -B $BUILD \
+    -D ENABLE_EXTRA_ALIGNMENT=FALSE \
+    -D ENABLE_AUTOMATIC_INIT_AND_CLEANUP=FALSE \
+    -D CMAKE_BUILD_TYPE=RelWithDebInfo \
+    -D ENABLE_MONGOC=FALSE
+
+
+If all dependencies are satisfied, the above command should succeed and end
+with::
+
+  $ cmake …
+  ## … (Lines of output) …
+  -- Generating done
+  -- Build files have been written to: $BUILD
+
+If configuration failed with an error, refer to the CMake output for error
+messages and information. Ensure that configuration succeeds before proceeding.
+
+.. dropdown:: What do these CMake arguments mean?
+  :class-container: admonition hint
+  :class-title: admonition-title
+  :animate: fade-in
+
+  The `ENABLE_EXTRA_ALIGNMENT` and `ENABLE_AUTOMATIC_INIT_AND_CLEANUP` are part
+  of |mongo-c-driver|, and correspond to deprecated features that are only
+  enabled by default for ABI compatibility purposes. It is highly recommended to
+  disable these features whenever possible.
+
+  The `ENABLE_MONGOC=FALSE` argument disabled building |libmongoc|. We'll build
+  that in the next section.
+
+  The |cmvar:CMAKE_BUILD_TYPE| setting tells CMake what variant of code will be
+  generated. In the case of `RelWithDebInfo`, optimized binaries will be
+  produced, but still include debug information. The |cmvar:CMAKE_BUILD_TYPE| has no
+  effect on Multi-Config generators (i.e. Visual Studio), which instead rely on
+  the `--config` option when building/installing.
+
+.. _CMAKE_BUILD_TYPE: https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
+
+
+.. _learn.get.build:
+
+Building the Project
+********************
+
+After successfully configuring the project, the build can be executed by using
+CMake::
+
+  $ cmake --build $BUILD --config RelWithDebInfo --parallel
+
+If configured properly and all dependencies are satisfied, then the above
+command should proceed to compile and link the configured components. If the
+above command fails, then there is likely an error with your environment, or you
+are using an unsupported/untested platform. Refer to the build tool output for
+more information.
+
+.. dropdown:: The ``--config`` option
+  :class-container: admonition hint
+  :class-title: admonition-title
+  :animate: fade-in
+
+  The :option:`--config <cmake--build.--config>` option is used to set the build
+  configuration to use in the case of Multi-Config generators (i.e. Visual
+  Studio). It has no effect on other generators, which instead use
+  |cmvar:CMAKE_BUILD_TYPE|.
+
+.. _learn.get.install:
+
+Installing the Built Results
+****************************
+
+Let `$PREFIX` be the path `$SOURCE/_install`. We can use CMake to install the
+built results::
+
+  $ cmake --install $BUILD --prefix "$PREFIX" --config RelWithDebInfo
```

Comment:
Can/should `cmake --build $BUILD --target install` be used instead of `cmake --install`? That seems to bypass the need to build separately.

Reply:
The `--install` CMake subcommand was added to replace invoking an "install" target, since the `install` target is not defined on all generators and `--install` gives more fine-grained controls.